src/validation.cpp

```diff
@@ -5302,3 +5302,18 @@ Chainstate& ChainstateManager::ActivateExistingSnapshot(CTxMemPool* mempool, uin
     m_active_chainstate = m_snapshot_chainstate.get();
     return *m_snapshot_chainstate;
 }
+
+const CBlockIndex* ChainstateManager::GetSnapshotBaseBlock() const
+{
+    AssertLockHeld(::cs_main);
+    const auto blockhash_op{this->SnapshotBlockhash()};
+    if (!blockhash_op) return nullptr;
+    return m_blockman.LookupBlockIndex(*blockhash_op);
```

Comment:
That seems like a good idea, also because `GetSnapshotBaseHeight` will deference the result to get a height. (Though it won't crash because it checks if the result is a `nullptr`.)